Sources/MapboxNavigation/NavigationLocationProvider.swift

```diff
@@ -0,0 +1,108 @@
+import Foundation
+import CoreLocation
+import MapboxCoreNavigation
+import MapboxMaps
+
+/**
+ An object that notifies a map view when the user’s location changes, minimizing the noise that normally accompanies location updates from a `CLLocationManager` object.
+ 
+ If your application displays a `MapView` before starting turn-by-turn navigation, call `LocationManager.overrideLocationProvider(with:)` to override default location provider so that the map view always shows the location snapped to the road network. For example, use this class to show the user’s current location as they wander around town.
+ 
+ This class depends on `NavigationLocationManager` to detect the user’s location as it changes.
+ */
+open class NavigationLocationProvider: NSObject, LocationProvider, CLLocationManagerDelegate {
+    /**
+     The location provider's location manager, which detects the user’s location as it changes.
+     */
+    public var locationManager: NavigationLocationManager
+    
+    public var locationProviderOptions: LocationOptions {
+        didSet {
+            locationManager.distanceFilter = locationProviderOptions.distanceFilter
+            locationManager.desiredAccuracy = locationProviderOptions.desiredAccuracy
+            locationManager.activityType = locationProviderOptions.activityType
+        }
+    }
+    
+    /**
+     The location provider's delegate.
+     */
+    public weak var delegate: LocationProviderDelegate?
+    
+    public var authorizationStatus: CLAuthorizationStatus {
+        CLLocationManager.authorizationStatus()
+    }
+    
+    public var accuracyAuthorization: CLAccuracyAuthorization {
+        if #available(iOS 14.0, *) {
+            return locationManager.accuracyAuthorization
+        } else {
+            return .fullAccuracy
+        }
+    }
+    
+    public var heading: CLHeading? {
+        return locationManager.heading
+    }
+    
+    public var headingOrientation: CLDeviceOrientation {
+        get {
+            locationManager.headingOrientation
+        }
+        set {
+            locationManager.headingOrientation = newValue
+        }
+    }
+
+    /**
+     Initializes the location provider with the `NavigationLocationManager`.
+     */
+    public override init() {
+        locationManager = NavigationLocationManager()
```

Comment:
If NavigationLocationProvider is intended to be the way to adapt a ReplayLocationManager or SimulatedLocationManager to a map view, then we’d need a convenience initializer that can set `locationManager` to a passed-in instance of NavigationLocationManager instead of creating a new one automatically.

Reply:
Thanks for the suggestion, I'll add a `locationManager` as the parameter for the initialization.